src/ServiceInsight/Explorer/EndpointExplorer/EndpointExplorerViewModel.cs

```diff
@@ -168,21 +168,24 @@ public async Task ConnectToService(string url)
             }
 
             var available = false;
-            using (workNotifier.NotifyOfWork($"Verifying ServiceControl availability at {url}"))
+            var address = url;
+
+            using (workNotifier.NotifyOfWork($"Verifying ServiceControl availability at {address}"))
             {
-                clientRegistry.EnsureServiceControlClient(url);
-                var serviceControl = clientRegistry.GetServiceControl(url);
-                available = await serviceControl.IsAlive();
+                var serviceControl = clientRegistry.Create(url);
+                (available, address) = await serviceControl.IsAlive();
             }
 
             if (!available)
             {
                 return;
             }
 
-            using (workNotifier.NotifyOfWork($"Connecting to ServiceControl at {url}"))
+            clientRegistry.EnsureServiceControlClient(address);
+
+            using (workNotifier.NotifyOfWork($"Connecting to ServiceControl at {address}"))
             {
-                var node = AddServiceNode(url);
+                var node = AddServiceNode(address);
```

Comment:
Does this mean that the customer is going to see the redirected url in their list of saved options and not the one they entered? It's not necessarily an issue, I just wanted to make sure it is intended..

Reply:
yes, that's how it'll be. This will treat the two connections as one.